src/main/scala/com/mwz/sonar/scala/scoverage/ScoverageSensor.scala

```diff
@@ -0,0 +1,160 @@
+              saveComponentScoverage(context, file, fileCoverage.fileScoverage)
+
+              //save the coverage of each line of the file
+              val coverage = context.newCoverage()
+              coverage.onFile(file)
+              for ((linenum, hits) <- fileCoverage.linesCoverage) {
+                coverage.lineHits(linenum, hits)
+              }
+              coverage.save()
+            }
+            case None => {
+              logger.warn(
+                s"[scoverage] The file: '${filename}', has no scoverage information associated to it."
+              )
+            }
+          }
+        }
+      }
+      case Failure(ex) => {
+        logger.error(
+          s"""[scoverage] Aborting the scoverage sensor execution,
+             |cause: an error occurred while reading the scoverage report file: '${reportFilename}',
+             |the error was: ${ex.getMessage}.""".stripMargin
+        )
+      }
+    }
+  }
+
+  /** Returns all scala main files from this module */
+  private[this] def getModuleSourceFiles(fs: FileSystem): Iterable[InputFile] = {
+    val predicates = fs.predicates
+    val predicate = predicates.and(predicates.hasLanguage(Scala.Key), predicates.hasType(InputFile.Type.MAIN))
+    fs.inputFiles(predicate).asScala
+  }
+
+  /** Returns the filename of the scoverage report for this module */
+  private[this] def getScoverageReportFilename(settings: Configuration): String = {
+    settings
+      .get(ScoverageSensorInternal.ScoverageReportPathPropertyKey)
+      .toOption
+      .getOrElse(
+        ScoverageSensorInternal.getDefaultScoverageReportPath(Scala.getScalaVersion(settings))
+      )
+  }
+
+  /** Saves the [[ScoverageMetrics]] of a component */
+  private[this] def saveComponentScoverage(
+    context: SensorContext,
+    component: InputComponent,
+    scoverage: Scoverage
+  ): Unit = {
+    context
+      .newMeasure[java.lang.Integer]()
+      .on(component)
+      .forMetric(ScoverageMetrics.totalStatements)
+      .withValue(scoverage.totalStatements)
+      .save()
+
+    context
+      .newMeasure[java.lang.Integer]()
+      .on(component)
+      .forMetric(ScoverageMetrics.coveredStatements)
+      .withValue(scoverage.coveredStatements)
+      .save()
+
+    context
+      .newMeasure[java.lang.Double]()
+      .on(component)
+      .forMetric(ScoverageMetrics.statementCoverage)
+      .withValue(scoverage.statementCoverage)
+      .save()
+
+    context
+      .newMeasure[java.lang.Double]()
+      .on(component)
+      .forMetric(ScoverageMetrics.branchCoverage)
+      .withValue(scoverage.branchCoverage)
+      .save()
+  }
+}
+
+object ScoverageSensorInternal {
+  private val SensorName = "Scoverage Sensor"
+  private val ScoverageReportPathPropertyKey = "sonar.scala.scoverage.reportPath"
+  private def getDefaultScoverageReportPath(scalaRawVersion: String) = {
+    //remove the extra part of the scala version, e.g 2.11.0 -> 2.11
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Space after the start of the comment [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.sonar-scala.com/coding_rules#rule_key=Scalastyle%3Ascalastyle_SpaceAfterCommentStartChecker)